pkg/operator/controller/garden/care/health.go

```diff
@@ -253,18 +259,18 @@ func (h *GardenHealth) checkObservabilityComponents(ctx context.Context, checker
 	return &c, nil
 }
 
-func (h *GardenHealth) isVPAEnabled() bool {
+func (h *Health) isVPAEnabled() bool {
 	return h.garden.Spec.RuntimeCluster.Settings != nil &&
 		h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler != nil &&
 		pointer.BoolDeref(h.garden.Spec.RuntimeCluster.Settings.VerticalPodAutoscaler.Enabled, false)
 }
 
-func checkManagedResourceForGarden(checker *HealthChecker, condition gardencorev1beta1.Condition, managedResource *resourcesv1alpha1.ManagedResource) *gardencorev1beta1.Condition {
+func checkManagedResourceForGarden(checker *care.HealthChecker, condition gardencorev1beta1.Condition, managedResource *resourcesv1alpha1.ManagedResource, clock clock.Clock) *gardencorev1beta1.Condition {
```

Comment:
As you have removed the specific parts at other places, do you also want to drop it here?

```suggestion
func checkManagedResource(checker *care.HealthChecker, condition gardencorev1beta1.Condition, managedResource *resourcesv1alpha1.ManagedResource, clock clock.Clock) *gardencorev1beta1.Condition {
```

Reply:
Good point 👍  I already continued working on another PR which streamlines more health related functionality. There, this function is even dropped (see https://github.com/timuthy/gardener/commit/21f6d7f734fc9eb3719bb2d97ba7fd090703dce4). Is it fine for you to keep it in this PR and contribute the change as part of a new one?
The same is true for https://github.com/gardener/gardener/pull/8378#discussion_r1301696743.

Reply:
Works for me.

Reply:
Decided to add it through [884f6fb](https://github.com/gardener/gardener/pull/8378/commits/884f6fb40565e1ea9d66d231121c29d389eef8e9).